### ISSUE TITLE
Support configuration of `typing` vs `typing_extensions`

### ISSUE THREAD

**Issue by tobni** [youtype/mypy_boto3_builder#221]
**Describe your idea**
I would like to enforce the library to use `typing_extensions`, regardless of my interpreter version.
 The top level `if` clause in the stub files could maybe support some type of configuration, like an environment variable?
**Code sample**
```python
if sys.version_info >= (3, 9) or not os.environ.get("BOTO_STUBS_USE_TYPING_EXTENSIONS"):
    from typing import TypedDict
else:
    from typing_extensions import TypedDict
```

**Additional context**
`pydantic` v2 requires `TypedDict` to be `typing_extensions.TypedDict` unless python is version 3.12 or higher. I cannot upgrade to this version of python, but I do want pydantics `TypedDict` support for parsing your generated `TypedDict`s.

**Comment by vemel** [comment]
Hello,

Thank you for the feature request. It makes sense. However, env variables are not supported in `*.pyi` files, so it will not work. The solution could be to use `typing_extensions` if available, and fallback to `typing`. Another solution would be always to use `typing_extensions` and make it a mandatory dependency.

I will check `pydantic` docs.

**Comment by tobni** [comment]
I was under the impression that there are both `type_defs.py` and `type_defs.pyi`? I think only the former matters for pydantic, as it is performing inspections at runtime.

**Comment by tobni** [comment]
https://docs.pydantic.dev/2.3/errors/usage_errors/#typed-dict-version is the cause for my request.

**Comment by vemel** [comment]
I just released `mypy_boto3_builder 7.18.0` with new import logic. I am going to generate all packages later today.

**Comment by vemel** [comment]
@tobni I released all `boto3-stubs` packages built with the new builder version. Please test and let me know if it works as it should.
